session/src/main/java/org/apache/iotdb/session/pool/SessionPool.java

```diff
@@ -420,6 +555,43 @@ public void insertRecord(String deviceId, long time, List<String> measurements,
     }
   }
 
+  /**
+   * insert data in one row asynchronously, if you want improve your performance,
+   * please use insertRecords method or insertTablet method
+   *
+   * @see Session#insertRecords(List, List, List, List, List)
+   * @see Session#insertTablet(Tablet)
+   */
+  public void asyncInsertRecord(String deviceId, long time, List<String> measurements,
+      List<String> values) {
+    CompletableFuture<Void> timeout = failAfter(Duration.ofSeconds(ASYNC_TIMEOUT));
+    CompletableFuture<Void> asyncRun = CompletableFuture.supplyAsync(() -> {
+      try {
+        insertRecord(deviceId, time, measurements, values);
+      } catch (IoTDBConnectionException | StatementExecutionException e) {
+        logger.error("Error occurred when inserting tablets: ", e);
+      }
+      return null;
+    }, threadPool.getThreadPool());
+
+    asyncRun.acceptEither(timeout, this::asyncHandler);
+  }
+
+  private static <T> CompletableFuture<T> failAfter(Duration duration) {
+    ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
+
+    CompletableFuture<T> promise = new CompletableFuture<>();
+    scheduler.schedule(() -> {
+      final TimeoutException ex = new TimeoutException("Timeout after " + duration);
+      return promise.completeExceptionally(ex);
+    }, duration.toMillis(), TimeUnit.MILLISECONDS);
+    return promise;
+  }
+
+  private void asyncHandler(Void aVoid) {
+    logger.info("Insertion executed successfully.");
```

Comment:
change to debug level and print the device + time

Reply:
Fixed